frontend/src/entities/group/create/group-create.tsx

```diff
@@ -0,0 +1,217 @@
+import Form from '@cloudscape-design/components/form';
+import ContentLayout from '../../../shared/layout/content-layout';
+import { createGroup, getGroup, updateGroup } from '../group.reducer';
+
+export type GroupCreateProperties = {
+    isEdit?: boolean;
+};
+
+export function GroupCreate ({isEdit}: GroupCreateProperties) {
+    const dispatch = useAppDispatch();
+    const notificationService = NotificationService(dispatch);
+    const navigate = useNavigate();
+    const {groupName} = useParams();
+    const [initialLoaded, setInitialLoaded] = useState(false);
+
+    const groupSchema = z.object({
+        name: z
+            .string()
+            .min(3, {
+                message: 'Group name must be longer than 3 characters',
+            })
+            .max(24, {
+                message: 'Group name cannot be more than 24 characters',
+            })
+            .regex(/^[a-zA-Z0-9]+$/, {
+                message: 'Group name can only contain alphanumeric characters.',
+            }),
+        description: z
+            .string({
+                invalid_type_error: 'Group description must be a string',
+            })
+            .min(1, {
+                message: 'Group description cannot be empty',
+            })
+            .max(4000, {
+                message: 'Group description cannot be more than 4000 characters.',
+            })
+            .regex(/[ -~]/, {
+                message: 'Group description can only contain printable characters',
+            }),
+    });
+
+    const {state, setState, setFields, touchFields} = useValidationReducer(groupSchema, {
+        validateAll: false,
+        needsValidation: false,
+        form: {
+            name: '',
+            description: '',
+        },
+        touched: {},
+        formErrors: {} as any,
+        formSubmitting: false,
+    });
+
+    useMemo(() => {
+        if (isEdit && !initialLoaded) {
+            setState({formSubmitting: true});
+            dispatch(getGroup(groupName)).then((response) => {
+                if (response.payload) {
+                    setInitialLoaded(true);
+                    setState({
+                        form: {
+                            name: response.payload?.data.group?.name || '',
+                            description: response.payload?.data.group?.description || '',
+                        },
+                        formSubmitting: false
+                    });
+                } else {
+                    navigate('/404');
+                }
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isEdit, groupName, initialLoaded]);
+
+
+    state.formErrors = {};
+    const parseResult = groupSchema.safeParse(state.form);
+    if (!parseResult.success) {
+        state.formErrors = issuesToErrors(
+            parseResult.error.issues,
+            state.validateAll === true ? undefined : state.touched
+        );
+    }
```

Comment:
This block is probably unnecessary. The `useValidationReducer` function above also runs `isValid` and `errors` and is essentially doing [the same thing](https://github.com/awslabs/mlspace/blob/main/frontend/src/shared/validation/index.ts#L296-L303) inside the method

Reply:
Thanks, I hadn't used this before. Still learning the libraries.